src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -318,7 +320,8 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 
 register_cli_argument('network nic create', 'enable_ip_forwarding', options_list=('--ip-forwarding',), help='Enable IP forwarding.', action='store_true')
 register_cli_argument('network nic create', 'network_interface_name', nic_type, options_list=('--name', '-n'), id_part=None, validator=process_nic_create_namespace)
-register_cli_argument('network nic create', 'private_ip_address_version', help='The private IP address version to use.')
+if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-09-01'):
+    register_cli_argument('network nic create', 'private_ip_address_version', help='The private IP address version to use.', default=IPVersion.ipv4.value)
```

Comment:
Good point.  In that case, please add the else... ignore_type statements so that the parameter doesn't appear in unsupported API versions.